lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -347,6 +347,12 @@ def create_post_for_auth_or_purchase(money, payment, options)
           add_creditcard(post, payment, options)
         end
 
+        add_charge_details(post, money, payment, options)
+        post
+      end
+
+      # Used internally by Spreedly to populate the charge object for 3DS 1.0 transactions
+      def add_charge_details(post, money, payment, options)
```

Comment:
I plan to simplify the code in Core by calling this method, rather than duplicating the logic. I felt a comment was necessary to indicate this shouldn't be renamed/removed.

Reply:
Just curious, because I was getting confused trying to understand the existing code. Because `create_source` it's not called internally in the class (unless I missed it), I am not 100% why it's a private method. Is `create_source` also a Spreedly-specific method?

This isn't really a review comment as much as a question as I try to understand what's going on. Still reviewing the PR.

Reply:
@bayprogrammer Yes, it is Spreedly specific. Our implementation of 3DS 1.0 uses the sources API. As you might have gathered from this PR, using a `source` with the `charges` API is different than the gateway adapter implementation of a charge. I was not the original author of `create_source`, but I can imagine it was a conscious decision to make it private.

https://github.com/spreedly/core/blob/master/app/models/gateways/stripe_gateway.rb#L88